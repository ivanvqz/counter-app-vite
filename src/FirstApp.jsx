import PropTypes from 'prop-types' //npm i prop-types
// const getSaludo = ( nombre, edad ) => 
//     `Saludos ${ nombre }, tienes ${edad} años`

const FirstApp = ( { title, subTitle, title2 } ) => {
    // Si no existen las properties se lanza un error
    
    return (
        <div className="container center">
            {/* <code>{ JSON.stringify( saludo ) }</code> */}
            {/* <h1> { getSaludo('Ivan', 23) } </h1> */}
            <h3 >{title2}</h3>
            <h2>{subTitle}</h2>
            <h2>{subTitle}</h2>
            <h1 data-testid="test-subtitle"> {title} </h1>
        </div>
    )
}
// para saber cómo funciona el componente, al menos en las properties
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired
}
//Valores por defecto
FirstApp.defaultProps = {
    title: 'No hay título',
    // subTitle: 'No hay subtítulo'
}

export default FirstApp