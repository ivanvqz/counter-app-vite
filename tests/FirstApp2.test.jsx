import { render, screen } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => {     
    const title = 'me llamo ivan';
    const subTitle = 'Soy desarrollador web';

    test('Debe de hacer match con snapshot', () => { 
        const { container } = render( <FirstApp subTitle={title} /> )
        expect( container ).toMatchSnapshot()
    })

    test('Debe de mostrar el mensaje "me llamo ivan" ', () => { 
        render( <FirstApp subTitle={ title } /> )
        //screen: es el objeto que renderizamos
        expect( screen.getAllByText( title ) ).toBeTruthy()
        screen.debug()
    })

    test('Debe de mostrar un pÃ¡rrafo en un p', () => { 
        render( <FirstApp title={ title } /> )
        //el role es el tipo de elemento que es
        expect( screen.getByRole( 'heading', {level: 1} ).innerHTML ).toContain( title )
    })

    test('debe de mostrar el subtitulo mostrado por props', () => { 
        render( <FirstApp title={ title } subTitle={ subTitle } /> )

        expect( screen.getAllByText( subTitle ).length).toBe(2)
    })
})