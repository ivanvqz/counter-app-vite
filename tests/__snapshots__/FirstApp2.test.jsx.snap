// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con snapshot 1`] = `
<div>
  <div
    class="container center"
  >
    <h3 />
    <h2>
      me llamo ivan
    </h2>
    <h2>
      me llamo ivan
    </h2>
    <h1
      data-testid="test-subtitle"
    >
       
      No hay t√≠tulo
       
    </h1>
  </div>
</div>
`;
